### PR TITLE
(F4xx ADC Support) I don't know what the HAL is going on

### PR DESCRIPTION
**Pull request by tmb5932**
F4xx ADC Support!!! Hooray!

Credits to Rue for the title :)

### CODE REVIEW

File: src/io/platform/f4xx/ADCf4xx.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
+/**
+ * DISCLAIMER: THIS MIGHT BREAK AT ANY POINT AND/OR NOT WORK FOR CERTAIN PURPOSES
```
Comment (mjmagee991) [inline src/io/platform/f4xx/ADCf4xx.cpp:2]: <3

File: src/io/platform/f4xx/ADCf4xx.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+    halADC.Init.DataAlign = ADC_DATAALIGN_RIGHT;
+    halADC.Init.NbrOfConversion = num_channels;
+    halADC.Init.DMAContinuousRequests = ENABLE;
+    halADC.Init.EOCSelection = ADC_EOC_SEQ_CONV;
+
+    __HAL_RCC_ADC1_CLK_ENABLE();
+    HAL_ADC_Init(&halADC);
+}
+
+void ADCf4xx::initDMA() {
```
Comment (mjmagee991) [inline src/io/platform/f4xx/ADCf4xx.cpp:105]: You deleted a TODO here. Do you understand what issue it was bringing up?

**Review by mjmagee991 — APPROVED** [review]
Honestly not much to comment on here. I just had a question about the deleted TODO related to DMA. Nice work!

File: src/io/platform/f4xx/ADCf4xx.cpp (hunk 2 above)
Comment (tmb5932) [reply src/io/platform/f4xx/ADCf4xx.cpp:105]: I believe the todo here was find out why there was commented out code in the f3xx ADC. 
The commented code was stopping the non-DMA ADC before initializing the DMA, but that was redundant and possibly harmful as right before initDMA is called it stops DMA ADC if it was on.

**Pull request merged**